packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -447,69 +734,290 @@ class _CupertinoAlertActionSection extends StatelessWidget {
+        isStacked: true,
+        children: <Widget>[
+          new BaseLayoutId<_CupertinoDialogActionsRenderWidget, MultiChildLayoutParentData>(
+            id: _AlertDialogSections.actionsSection,
+            child: actionsSection,
+          ),
+        ]..addAll(buttons),
+      );
     } else {
       // For a horizontal layout, we don't need the scrollController in most
       // cases, but it still has to be always attached to a scroll view.
-      return new CupertinoScrollbar(
+      final Widget actionsSections = new CupertinoScrollbar(
         child: new SingleChildScrollView(
           controller: scrollController,
-          child: new CustomPaint(
-            painter: new _CupertinoHorizontalDividerPainter(children.length),
-            child: new UnconstrainedBox(
-              constrainedAxis: Axis.horizontal,
-              child: new ConstrainedBox(
-                constraints: const BoxConstraints(minHeight: _kButtonHeight),
-                child: new Row(
-                  children: children.map<Widget>((Widget button) {
-                    return new Expanded(child: button);
-                  }).toList(),
-                ),
+          child: ConstrainedBox(
+            constraints: new BoxConstraints.loose(
+              const Size(
+                _kCupertinoDialogWidth,
+                double.infinity,
               ),
             ),
+            child: _buildHorizontalButtons(devicePixelRatio),
           ),
         ),
       );
+
+      return new _CupertinoDialogActionsRenderWidget(
+        isStacked: false,
+        children: <Widget>[
+          new BaseLayoutId<_CupertinoDialogActionsRenderWidget, MultiChildLayoutParentData>(
+            id: _AlertDialogSections.actionsSection,
+            child: actionsSections,
+          ),
+          actionsSections,
+        ],
+      );
     }
   }
 }
 
-// A CustomPainter to draw the divider lines.
+// iOS style layout policy widget for sizing action buttons.
+//
+// The sizing policy is partially determined by whether or not action buttons
+// are stacked vertically, or positioned horizontally. [isStacked] is used to
+// indicate whether or not the buttons should be stacked vertically.
+//
+// See [_CupertinoDialogActionsRenderBox] for specific layout policy details.
+//
+// Usage instructions:
+//
+// When stacked vertically:
+// The entire actions section (all buttons and dividers) should be a single
+// grandchild of this widget, and it should be wrapped with a
+// [BaseLayoutId<_CupertinoDialogActionsRenderWidget, MultiChildLayoutParentData>]
+// whose ID is [_AlertDialogPieces.actionsSection].
+//
+// Also, the entire list of buttons and dividers should also be passed as
+// direct children of this widget in the order they appear: divider, button,
+// divider, button. The order is critical and the layout will break if that
+// exact order is not respected.
+//
+// Vertical example:
+//
+// ```
+// new _CupertinoDialogActionsRenderWidget(
```

Comment:
sample code should be in a \`\`\`dart block after a `## Sample code` header, so that it gets analyzed.

Reply:
Is this true even in non-public documentation?

Reply:
oh, i didn't realise this was just a comment. never mind.